### PR TITLE
Do some improvements on the spdx-tools wrapper script

### PR DESCRIPTION
**Pull request by maxhbr**
* now runs, even if `JAVA_HOME` is not set
* runs on all cores, not just pinned to 8
* write a way less complex inline bash command

still open (but out of scope):
- [x] with parallel execution, the log is pretty unusable
  - for PRs this is solved. For main runs this is not so important

### CODE REVIEW

**Review by OliverFendt — APPROVED** [review]
@maxhbr thank you very much. LGTM

**Comment by heliocastro** [comment]
@OliverFendt @maxhbr I would appreciate if you do no wipe my credits on oth files, and put Max as the only author of this, ok ?

**Comment by maxhbr** [comment]
sorry, I created https://github.com/Open-Source-Compliance/package-analysis/pull/28

File: .github/workflows/scripts/spdx-tools-java-wrapper.sh [hunk 1]
```diff
@@ -8,42 +8,54 @@
...
 jar="$HOME/spdx-tools-java/tools-java-${version}-jar-with-dependencies.jar"
 
-if [[ -n "$JAVA_HOME" && -x "$JAVA_HOME/bin/java" ]]; then
-    javabin="$JAVA_HOME/bin/java"
-else
-    javabin=java
-fi
+spdx_tools() {
+    local javabin
+    if [[ -n "${JAVA_HOME+x}" && -x "$JAVA_HOME/bin/java" ]]; then
```
Comment (sschuberth) [inline .github/workflows/scripts/spdx-tools-java-wrapper.sh:15]: In modern Bash, shouldn't that be `-v $JAVA_HOME` instead?

File: .github/workflows/scripts/spdx-tools-java-wrapper.sh [hunk 2]
```diff
@@ -8,42 +8,54 @@
...
-zip="tools-java-${version}.zip"
-url="https://github.com/spdx/tools-java/releases/download/v${version}/${zip}"
 jar="$HOME/spdx-tools-java/tools-java-${version}-jar-with-dependencies.jar"
 
-if [[ -n "$JAVA_HOME" && -x "$JAVA_HOME/bin/java" ]]; then
-    javabin="$JAVA_HOME/bin/java"
-else
-    javabin=java
-fi
+spdx_tools() {
```
Comment (sschuberth) [inline .github/workflows/scripts/spdx-tools-java-wrapper.sh:13]: Unfortunately, the commit message does not detail the very different changes done in this commit.

**Pull request merged**